ocs_ci/ocs/resources/pod.py

```diff
@@ -2263,3 +2263,101 @@ def get_crashcollector_pods(
     namespace = namespace or config.ENV_DATA["cluster_namespace"]
     crashcollectors = get_pods_having_label(crashcollector_label, namespace)
     return [Pod(**crashcollector) for crashcollector in crashcollectors]
+
+
+def check_pods_in_statuses(
+    expected_statuses,
+    pod_names=None,
+    namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+    raise_pod_not_found_error=False,
+):
+    """
+    checks whether the pods in a given namespace are in the expected statuses or not
+
+    Args:
+        expected_statuses (list): The expected statuses of the pods
+        pod_names (list): List of the pod names to check.
+            If not provided, it will check all the pods in the given namespace
+        namespace (str): Name of cluster namespace(default: defaults.ROOK_CLUSTER_NAMESPACE)
+        raise_pod_not_found_error (bool): If True, it raises an exception, if one of the pods
+            in the pod names are not found. If False, it ignores the case of pod not found and
+            check the pod objects of the rest of the pod names. The default value is False
+
+    Returns:
+        Boolean: True, if the pods are in the expected statuses. False, otherwise
+
+    """
+    ret_val = True
+
+    if pod_names:
+        list_of_pods = get_pod_objs(pod_names, raise_pod_not_found_error)
+    else:
+        list_of_pods = get_all_pods(namespace)
+
+    ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
+    for p in list_of_pods:
+        try:
+            status = ocp_pod_obj.get_resource(p.name, "STATUS")
+        except CommandFailed as e:
+            logger.info(f"Can't get the pod status due to the error: {str(e)}")
+            status = ""
+
+        if status not in expected_statuses:
+            logger.warning(
+                f"The pod {p.name} is in {status} state, and not in the expected statuses {expected_statuses}"
+            )
+            ret_val = False
+
+    logger.info(f"All the pods reached the expected statuses {expected_statuses}")
```

Comment:
This message will be printed also if `ret_val` will be set to `False`. There should be condition like this:
```suggestion
    if ret_val:
        logger.info(f"All the pods reached the expected statuses {expected_statuses}")
```

Or if it is not necessary to always check all the pods, if one of them is not in the expected state, then you can remove the `ret_val` variable and on line `2309` change `ret_val = False` to `return False` (and on the following line change `return ret_val` to `return True`).

Reply:
Yes, right. I will fix it.